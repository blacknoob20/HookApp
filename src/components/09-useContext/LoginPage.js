import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const LoginPage = () => {
    const { setUser } = useContext(UserContext);
    
    return (
        <div>
            <h1>Login Page</h1>
            <hr />
            <button className="btn btn-primary" onClick={() => { setUser({ id: 1234, name: 'Cristhian' }) }}>Login</button>
        </div>
    )
}
