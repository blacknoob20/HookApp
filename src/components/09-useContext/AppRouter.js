import React from 'react';
import { BrowserRouter as Router, Route, Switch, Redirect } from 'react-router-dom';
import { AboutPage } from './AboutPage';
import { HomePage } from './HomePage';
import { LoginPage } from './LoginPage';
import { NavBar } from './NavBar';

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar />
                <Switch>
                    <Route exact path="/about" component={AboutPage} />
                    <Route exact path="/login" component={LoginPage} />
                    <Route exact path="/" component={HomePage} />
                    <Redirect to="/" />
                    {/* <Route component={HomePage} /> */}
                </Switch>
            </div>
        </Router>
    )
}
