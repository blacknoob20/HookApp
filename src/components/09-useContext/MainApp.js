import React from 'react';
import { AppRouter } from './AppRouter';

export const MainApp = () => {
    return (
        <div>
            <h1>Main Page</h1>
            <hr />
            <AppRouter />
        </div>
    )
}
