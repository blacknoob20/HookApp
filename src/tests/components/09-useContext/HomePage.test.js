import { mount } from 'enzyme';
import { HomePage } from '../../../components/09-useContext/HomePage';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Pruebas en <HomePage/>', () => {
    const user = 'Cristhian';
    const email = 'reguerre@espol.edu.ec';
    const wrapper = mount(
        <UserContext.Provider value={user,email}>
            <HomePage />
        </UserContext.Provider>
    );

    test('Debe mostrarse correctamente.', () => {
        expect(wrapper).toMatchSnapshot();
    });

});
