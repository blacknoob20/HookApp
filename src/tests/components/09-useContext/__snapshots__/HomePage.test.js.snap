// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomePage/> Debe mostrarse correctamente. 1`] = `
<div>
  <h1>
    Home Page
  </h1>
  <hr />
  <pre />
</div>
`;
